src/group_theory/submonoid/basic.lean

```diff
@@ -428,3 +429,25 @@ add_decl_doc add_monoid_hom.of_mdense
 attribute [norm_cast] coe_of_mdense add_monoid_hom.coe_of_mdense
 
 end monoid_hom
+
+section pointwise
+
+namespace submonoid
+
+@[to_additive]
+lemma closure_mul_le (S T : set M) : closure (S * T) ≤ closure S ⊔ closure T :=
+Inf_le $ λ x ⟨s, t, hs, ht, hx⟩, hx ▸ (closure S ⊔ closure T).mul_mem
+    (le_def.mp le_sup_left $ subset_closure hs)
+    (le_def.mp le_sup_right $ subset_closure ht)
+
+@[to_additive]
+lemma sup_eq_closure (H K : submonoid M) : H ⊔ K = closure (H * K) :=
+le_antisymm
+  (sup_le
+    (λ h hh, subset_closure ⟨h, 1, hh, K.one_mem, mul_one h⟩)
+    (λ k hk, subset_closure ⟨1, k, H.one_mem, hk, one_mul k⟩))
+  (by conv_rhs { rw [← closure_eq H, ← closure_eq K] }; apply closure_mul_le)
+
+end submonoid
```

Comment:
This is causing the same cyclic import as described here:

https://github.com/leanprover-community/mathlib/blob/069ba435ed9c0387802cda6920b6b00e38d20454/src/algebra/pointwise.lean#L452-L476

Your choice are either to
* Hope the problem that comment describes no longer exists, and move the lemmas I link to to be next to your lemmas
* Assume it still exists, and move your lemmas next to those